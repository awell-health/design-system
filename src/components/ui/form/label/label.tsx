import { FC, ReactNode } from 'react';
import { cn } from '../../../../lib/utils';

interface Props {
  label: string | ReactNode | JSX.Element;
  sublabel?: string;
  className?: string;
  required?: boolean;
}

export const Label: FC<Props> = ({ label, sublabel, className, required }) => {
  return (
    <>
      <div className={cn('label', 'flex flex-col gap-1 items-start', className)}>
        <span className=' label-text text-slate-600 text-sm font-medium'>
          {label}
          {required && <span className='text-red-600 ml-0.5'>*</span>}
        </span>
        {sublabel && (
          <span className='label-text text-slate-400 text-xs font-normal'>{sublabel}</span>
        )}
      </div>
    </>
  );
};
