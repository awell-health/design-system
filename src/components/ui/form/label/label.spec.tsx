import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import React from 'react';
import { Label } from './label';

describe('Label', () => {
  it('match snapshot', () => {
    const { container } = render(<Label label='Label' sublabel='Sublabel' />);
    expect(container).toMatchSnapshot();
  });

  it('should render label and sublabel', () => {
    render(<Label label='Label' sublabel='Sublabel' />);
    expect(screen.getByText('Label')).toBeInTheDocument();
    expect(screen.getByText('Sublabel')).toBeInTheDocument();
  });

  it('should render asterisk when required', () => {
    render(<Label label='Label' sublabel='Sublabel' required />);
    expect(screen.getByText('*')).toBeInTheDocument();
  });
});
