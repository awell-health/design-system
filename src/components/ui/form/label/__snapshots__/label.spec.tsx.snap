// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > match snapshot 1`] = `
<div>
  <div
    class="label flex flex-col gap-1 items-start"
  >
    <span
      class=" label-text text-slate-600 text-sm font-medium"
    >
      Label
    </span>
    <span
      class="label-text text-slate-400 text-xs font-normal"
    >
      Sublabel
    </span>
  </div>
</div>
`;
